docs/proposals/ldap-group-sync.md

```diff
@@ -0,0 +1,105 @@
+# Syncing LDAP Groups to OpenShift Groups
+
+Linking OpenShift `Groups` to their representations on LDAP servers allows for administrators to have an easier way to manage `Groups` inside OpenShift. This additional functionality will require no additional fields on the OpenShift `Group` object. The sync job will indicate that an OpenShift `Group` mirrors an external representation by setting the `openshift.io/externalRecordProvider` `Annotation` in the `Group` as well as adding a `Label` such as `openshift.io/externalRecordProvider/LDAP` to the `Group`. Other information (e.g. last sync time or resource version) will be stored as `Annotations` on the OpenShift `Group`. Logistical information for the LDAP sync will be stored in an `LDAPSyncOptions` that the administrator will need to provide.
+
+### Use Cases
+1. I want to sync all LDAP groups I've configured in a sync job (e.g. a specific LDAP server configuration) to OpenShift `Groups`
+* I want to sync specific OpenShift `Groups` with their `externalRecordProviders`
+* I want to sync a speficic set of LDAP groups to OpenShift `Groups`
+* I want to run a destructive sync that will remove OpenShift groups that were created or updated by a previous sync if the LDAP server no longer contains their record
+* I want to map an LDAP group to an existing OpenShift `Group`
+
+## Supported LDAP Group Schemas
+The three LDAP schemas supported are
+
+1. groups as first-class LDAP entries that contain lists of members of the group
+2. users as first-class LDAP entries that contain lists of groups they are a member of
+3. schema 2 above, also including first-class groups to store metadata
+
```

Comment:
Add a section about required metadata in openshift groups:
labels (for filtering.  Can we check the existence of a label?)
 1. openshift.io/ldapURL = ldap url.  This is used for selection of pertinent openshift groups.

annotations
 1. openshift.io/ldapGroup = ldap groupUID
 1. openshift.io/ldapSyncTime = time of the last time this group was synced (everyone speak millis since the epoch, right?)